sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/HDFSBackedStateStoreProvider.scala

```diff
@@ -84,6 +84,13 @@ private[sql] class HDFSBackedStateStoreProvider extends StateStoreProvider with
       map.iterator()
     }
 
+    override def getWithCompositeKey(
```

Comment:
Not sure I understand why we need to update the store providers themselves ? Can't we just use an updated schema that encodes the required information within the key